### ISSUE TITLE
Add an "unlimited items" mode to wasm-smith

### ISSUE THREAD

**Issue by alexcrichton** [bytecodealliance/wasm-tools#180]
Wasm-smith is proving very useful at testing whether or not engines resource-constrain wasm modules.  For example it has expose https://github.com/bytecodealliance/wasm-tools/issues/179 as a weakness in validating module-linking modules.

I think it'd be useful to add a `SwarmConfig`-style generator for wasm-smith where items in the module itself are practically unlimited and may exceed implementation limits. That way we can test that wasmparser doesn't ever have a blowup of resources on these modules, nor does wasmtime itself blow up in resources. We would just need to thread through a boolean that the wasm-smith-generated-module may not be valid.

**Comment by fitzgen** [comment]
Can we not effectively just do this by increasing the `MAX_MAXIMUM`?

**Comment by alexcrichton** [comment]
Yeah that's basically what I'm thinking. Overall I think it would be interesting to add more modes to wasm-smith which aren't guaranteed to generate valid modules, but instead may generate invalid modules (as we do right now with function bodies, and with this perhaps with numbers of items as well).

I think this is possible with the `Config` trait today but it'd be nice if we had a built-in implementation of `Config` that did this perhaps.
